per-user-usage.py

```diff
@@ -1,118 +1,230 @@
+    for (thread_data.errorIndication,
+         thread_data.errorStatus,
+         thread_data.errorIndex,
+         thread_data.varBinds) in nextCmd(
+        SnmpEngine(),
+        auth_data,
+        SNMP_target,
+        ContextData(),
+        ObjectType(ObjectIdentity('.1.3.6.1.4.1.9.10.24.1.3.2.1.2.2')),
+        lexicographicMode=False
+        ):
+
+        if thread_data.errorIndication:
+            print(thread_data.errorIndication)
+            break
+        elif thread_data.errorStatus:
+            print(
+                '%s at %s' % (thread_data.errorStatus.prettyPrint(),
+                              thread_data.errorIndex and thread_data.varBinds[int(thread_data.errorIndex) - 1][0] or '?')
+                )
+            break
+        else:
+            for thread_data.varBind in thread_data.varBinds:
+                thread_data.uid = str(thread_data.varBind[0]).split(".")
+                if str(thread_data.varBind[1]) != "":
+                    circuit_ids[str(thread_data.uid[-2] + '.' + thread_data.uid[-1])] = str(thread_data.varBind[1])
+
+    print(str(len(circuit_ids)) + ' users found')
+
+
+def get_interface_ids(SNMP_target, auth_data, interface_IDs):
     thread_data = threading.local()
     print('Finding interface indexes...', flush=True)
     for (thread_data.errorIndication,
          thread_data.errorStatus,
          thread_data.errorIndex,
-         thread_data.varBinds) in nextCmd(SnmpEngine(),
-                              auth_data,
-                              SNMP_target,
-                              ContextData(),
-                              ObjectType(ObjectIdentity('.1.3.6.1.4.1.9.10.24.1.3.2.1.11')),
-                              lexicographicMode=False):
+         thread_data.varBinds) in nextCmd(
+        SnmpEngine(),
+        auth_data,
+        SNMP_target,
+        ContextData(),
+        ObjectType(ObjectIdentity('.1.3.6.1.4.1.9.10.24.1.3.2.1.11')),
+        lexicographicMode=False
+        ):
 
         if thread_data.errorIndication:
             print(thread_data.errorIndication)
             break
         elif thread_data.errorStatus:
-            print('%s at %s' % (thread_data.errorStatus.prettyPrint(),
-                                thread_data.errorIndex and thread_data.varBinds[int(thread_data.errorIndex) - 1][0] or '?'))
+            print(
+                '%s at %s' % (thread_data.errorStatus.prettyPrint(),
+                              thread_data.errorIndex and thread_data.varBinds[int(thread_data.errorIndex) - 1][0] or '?')
+                )
             break
         else:
             for thread_data.varBind in thread_data.varBinds:
                 thread_data.uid = str(thread_data.varBind[0]).split(".")
                 try:
-                    interface_IDs[str(thread_data.varBind[1])] = thread_data.uid[-2]+'.'+thread_data.uid[-1]
+                    interface_IDs[str(thread_data.varBind[1])] = str(thread_data.uid[-2] + '.' + thread_data.uid[-1])
```

Comment:
Why cast to str after + operator on strings?  It will be string type anyway, or file on int to str conversion.

Reply:
It should be, there was some weird behaviour for it that you wrapped in try and except statement. I wanted to investigate it further, but you can ignore this change.